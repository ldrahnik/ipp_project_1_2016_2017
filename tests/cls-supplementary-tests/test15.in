class A {

};
class C {
    protected:
        char32_t *question;
    public:
        int *myatt, test, uuu;
};
class B : public C {
    protected:
        static int the_answer(void);
    public:
        virtual bool is_real(int a, int b, int c) = 0;
};