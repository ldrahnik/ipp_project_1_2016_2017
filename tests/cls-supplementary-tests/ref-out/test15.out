<?xml version="1.0" encoding="UTF-8"?>
<class name="B" kind="abstract">
    <inheritance>
        <from name="C" privacy="public"/>
    </inheritance>
    <public>
        <attributes>
            <attribute name="myatt" type="int *" scope="instance">
                <from name="C"/>
            </attribute>
            <attribute name="test" type="int *" scope="instance">
                <from name="C"/>
            </attribute>
            <attribute name="uuu" type="int *" scope="instance">
                <from name="C"/>
            </attribute>
        </attributes>
        <methods>
            <method name="is_real" type="bool" scope="instance">
                <virtual pure="yes"/>
                <arguments>
                    <argument name="a" type="int"/>
                    <argument name="b" type="int"/>
                    <argument name="c" type="int"/>
                </arguments>
            </method>
        </methods>
    </public>
    <protected>
        <attributes>
            <attribute name="the_answer" type="int" scope="static"/>
            <attribute name="question" type="char32_t *" scope="instance">
                <from name="C"/>
            </attribute>
        </attributes>
        <methods>
            <method name="the_answer" type="int" scope="static">
                <from name=""/>
                <virtual pure="no"/>
                <arguments></arguments>
            </method>
        </methods>
    </protected>
</class>